apps/codelab/src/app/components/slides/title-slide/title-slide.component.ts

```diff
@@ -9,4 +12,36 @@ export class TitleSlideComponent {
   @Input() title: string;
   @Input() description: string;
   @Input() prereqs: string;
+
+  constructor(
+    private readonly router: Router,
+    private readonly presentation: SlidesDeckComponent,
+    @Optional() @Inject(MENU_ROUTES) readonly menuRoutes
+  ) {
+    if (this.presentation != null) {
+      this.setupPreviousNext();
+    }
+  }
+
+  private setupPreviousNext() {
+    let previousLink = '';
+    let nextLink = '';
+    let allRoutes = this.menuRoutes.map(p => p.path);
+    let currentUrl = this.router.url;
+    if (currentUrl.startsWith('/')) {
+      currentUrl = currentUrl.substr(1);
+    }
+    const urlPaths = currentUrl.split('/');
+    if (urlPaths.length > 1) {
+      const idx = allRoutes.indexOf(urlPaths[1]);
+      if (idx > 0) {
+        previousLink = `/${urlPaths[0]}/${allRoutes[idx - 1]}`;
+      }
+      if (idx < allRoutes.length - 1) {
+        nextLink = `/${urlPaths[0]}/${allRoutes[idx + 1]}`;
+      }
```

Comment:
This looks very complicated, is there some activeRoute property we could use to avoid all the  string manipulations?

Reply:
It is complicated for sure, if I can get or compute the full path from menu routes that will work too.